src/test/java/uk/ac/ebi/eva/pipeline/io/writers/VariantSourceEntityMongoWriterTest.java

```diff
@@ -86,15 +86,23 @@ public void shouldWriteAllFieldsIntoMongoDb() throws Exception {
         while (cursor.hasNext()) {
             count++;
             DBObject next = cursor.next();
-            assertTrue(next.get("fname") != null);
-            assertTrue(next.get("fid") != null);
-            assertTrue(next.get("sid") != null);
-            assertTrue(next.get("sname") != null);
-            assertTrue(next.get("samp") != null);
-            assertTrue(next.get("meta") != null);
-            assertTrue(next.get("stype") != null);
-            assertTrue(next.get("date") != null);
-            assertTrue(next.get("aggregation") != null);
+            assertNotNull(next.get("fname"));
+            assertNotNull(next.get("fid"));
+            assertNotNull(next.get("sid"));
+            assertNotNull(next.get("sname"));
+            assertNotNull(next.get("samp"));
+            assertNotNull(next.get("stype"));
+            assertNotNull(next.get("date"));
+            assertNotNull(next.get("aggregation"));
+
+            DBObject meta = (DBObject) next.get("meta");
+            assertNotNull(meta);
+            assertNotNull(meta.get("fileformat"));
+            assertNotNull(meta.get("header"));
```

Comment:
shouldn't we use constants here? at least the ones easily available as this one. One might say that the tests would be too coupled, but I think it would show where do those strings come from. A disadvantage of the current way, is that it may suggest that you should handle this object with hardcoded strings.

On a side note, I wonder whether the field names in VariantSourceEntity as "fname" can be retrieved with reflection.